face/FindFaces/python/add_face_to_collection.ipynb

```diff
@@ -0,0 +1,169 @@
+    "        face_rectangle = enlarge_bounding_box(faces[0]['faceRectangle'], image_width, image_height)\n",
+    "        params['targetFace'] = f\"{face_rectangle['left']},{face_rectangle['top']},{face_rectangle['width']},{face_rectangle['height']}\"\n",
+    "    else:\n",
+    "        print(f\"One face detected. Adding to the target.\")\n",
+    "\n",
+    "    if target_type == \"face_list\":\n",
+    "        # URL to add face to face list\n",
+    "        add_face_url = endpoint + f\"/face/v1.0/faceLists/{target_id}/persistedFaces\"\n",
+    "    elif target_type == \"large_face_list\":\n",
+    "        # URL to add face to large face list\n",
+    "        add_face_url = endpoint + f\"/face/v1.0/largefacelists/{target_id}/persistedFaces\"\n",
+    "    elif target_type == \"person_group\":\n",
+    "        if not person_id:\n",
+    "            return \"Person ID is required for adding to a person group.\"\n",
+    "        # URL to add face to person group\n",
+    "        add_face_url = endpoint + f\"/face/v1.0/persongroups/{target_id}/persons/{person_id}/persistedFaces\"\n",
+    "    elif target_type == \"large_person_group\":\n",
+    "        if not person_id:\n",
+    "            return \"Person ID is required for adding to a large person group.\"\n",
+    "        # URL to add face to large person group\n",
+    "        add_face_url = endpoint + f\"/face/v1.0/largepersongroups/{target_id}/persons/{person_id}/persistedFaces\"\n",
+    "    elif target_type == \"person_directory\":\n",
+    "        # URL to add face to person directory\n",
+    "        add_face_url = endpoint + f\"/face/v1.0-preview/persons/{target_id}/recognitionModels/recognition_04/persistedFaces\"\n",
+    "    else:\n",
+    "        return \"Invalid target type specified.\"\n",
+    "\n",
+    "    # Add the face to the specified target\n",
+    "    with open(image_path, 'rb') as image_data:\n",
+    "        response = requests.post(add_face_url, params=params, headers=headers, data=image_data)\n",
+    "        if response.status_code == 200:\n",
+    "            return \"Face added successfully.\"\n",
+    "        elif response.status_code == 202:\n",
+    "            operation_location = response.headers.get('Operation-Location')\n",
+    "            if operation_location:\n",
+    "                return check_operation_status(subscription_key, operation_location)\n",
+    "            else:\n",
+    "                print(\"No Operation-Location header found in the response.\")\n",
+    "        else:\n",
+    "            return f\"Failed to add face: {response.json()}\"\n",
+    "\n",
+    "# Example usage\n",
+    "FACE_KEY = os.environ[\"FACE_API_KEY\"]\n",
+    "FACE_ENDPOINT = os.environ[\"FACE_ENDPOINT_URL\"]\n",
+    "image_path = 'path_to_image.jpg'\n",
+    "target_type = 'face_list'  # or 'large_face_list', 'person_group', 'large_person_group', or 'person_directory'\n",
```

Comment:
Q: Do we really want to support "large_face_list", "large_person_group" or " person_directory" here? These collections are not demonstrated in the other three sample.

Reply:
> Q: Do we really want to support "large_face_list", "large_person_group" or " person_directory" here? These collections are not demonstrated in the other three sample.

Good question. This sample code is separate and will be shared with our customers who encountered error 400 when adding a face to our face collection. Therefore, it's better to support all related APIs. Please let me know your thoughts. Thanks.

Reply:
Thanks for the context. I’m with you on that.